### ISSUE TITLE
error: (-215:Assertion failed) top >= 0 && bottom >= 0 && left >= 0 && right >= 0 && _src.dims() <= 2 in function 'cv::copyMakeBorder'

### ISSUE THREAD

**Issue by dchatel** [ltdrdata/ComfyUI-Impact-Pack#329]
As of today I get this error on a workflow that had no issue at all with it before.
```
Error occurred when executing FaceDetailer:

OpenCV(4.7.0) D:\a\opencv-python\opencv-python\opencv\modules\core\src\copy.cpp:1026: error: (-215:Assertion failed) top >= 0 && bottom >= 0 && left >= 0 && right >= 0 && _src.dims() <= 2 in function 'cv::copyMakeBorder'


File "G:\ai\ComfyUI\execution.py", line 153, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "G:\ai\ComfyUI\execution.py", line 83, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "G:\ai\ComfyUI\execution.py", line 76, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "G:\ai\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\impact_pack.py", line 430, in doit
enhanced_img, cropped_enhanced, cropped_enhanced_alpha, mask, cnet_pil_list = FaceDetailer.enhance_face(
File "G:\ai\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\impact_pack.py", line 382, in enhance_face
segs = bbox_detector.detect(image, bbox_threshold, bbox_dilation, bbox_crop_factor, drop_size, detailer_hook=detailer_hook)
File "G:\ai\ComfyUI\custom_nodes\ComfyUI-Inspire-Pack\inspire\segs_support.py", line 37, in detect
segs = seg_obj().doit(facemesh_image, crop_factor, not self.is_segm, crop_min_size, drop_size, dilation,
File "G:\ai\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\segs_nodes.py", line 912, in doit
result = core.mediapipe_facemesh_to_segs(image, crop_factor, bbox_fill, crop_min_size, drop_size, dilation, face, mouth, left_eyebrow, left_eye, left_pupil, right_eyebrow, right_eye, right_pupil)
File "G:\ai\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 1042, in mediapipe_facemesh_to_segs
create_seg('face')
File "G:\ai\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 1037, in create_seg
seg = mask_to_segs(mask, False, crop_factor, bbox_fill, drop_size=drop_size, label=label, crop_min_size=crop_min_size)
File "G:\ai\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 948, in mask_to_segs
contours, _ = cv2.findContours(mask_i_uint8, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
```

**Comment by ltdrdata** [comment]
Update `opencv-python` version to latest version and try again.

**Comment by 4thanks** [comment]
same issue, and the OpenCV(4.8.1)  is the latest version, I just updated.
```
Error occurred when executing MediaPipeFaceMeshToSEGS:

OpenCV(4.8.1) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\contours.cpp:197: error: (-210:Unsupported format or combination of formats) [Start]FindContours supports only CV_8UC1 images when mode != CV_RETR_FLOODFILL otherwise supports CV_32SC1 images only in function 'cvStartFindContours_Impl'
```

**Comment by ltdrdata** [comment]
> same issue, and the OpenCV(4.8.1) is the latest version, I just updated.
> 
> ```
> Error occurred when executing MediaPipeFaceMeshToSEGS:
> 
> OpenCV(4.8.1) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\contours.cpp:197: error: (-210:Unsupported format or combination of formats) [Start]FindContours supports only CV_8UC1 images when mode != CV_RETR_FLOODFILL otherwise supports CV_32SC1 images only in function 'cvStartFindContours_Impl'
> ```

Send me the your workflow and sample input.

**Comment by 4thanks** [comment]
> Send me the your workflow and sample input.

![test-mediapie](https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/63891627/97cbc73c-c158-4780-aecb-3e51c06f5ed6)

**Comment by ltdrdata** [comment]
> > Send me the your workflow and sample input.
> 
> ![test-mediapie](https://private-user-images.githubusercontent.com/63891627/287141519-97cbc73c-c158-4780-aecb-3e51c06f5ed6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MDQ0NjgsIm5iZiI6MTcwMTQwNDE2OCwicGF0aCI6Ii82Mzg5MTYyNy8yODcxNDE1MTktOTdjYmM3M2MtYzE1OC00NzgwLWFlY2ItM2U1MWMwNmY1ZWQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDA0MTYwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4ZmMwZmNiMTk5ZWE5MzQ3NDhkMTcyYjYyZWQyODc5NTFiOGU2MTEyMDZiZDk5MWYzNDBiYjRhNmVlY2MxODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ylp4Uk5NoLvGy5D9eNakURF05vh1FrL583JvggIUbcw)

Please also upload the post-processed facemesh image below. It seems necessary to check if the issue is occurring at a specific resolution.

**Comment by 4thanks** [comment]
> > > Send me the your workflow and sample input.
> > 
> > 
> > ![test-mediapie](https://private-user-images.githubusercontent.com/63891627/287141519-97cbc73c-c158-4780-aecb-3e51c06f5ed6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MDQ0NjgsIm5iZiI6MTcwMTQwNDE2OCwicGF0aCI6Ii82Mzg5MTYyNy8yODcxNDE1MTktOTdjYmM3M2MtYzE1OC00NzgwLWFlY2ItM2U1MWMwNmY1ZWQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDA0MTYwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4ZmMwZmNiMTk5ZWE5MzQ3NDhkMTcyYjYyZWQyODc5NTFiOGU2MTEyMDZiZDk5MWYzNDBiYjRhNmVlY2MxODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ylp4Uk5NoLvGy5D9eNakURF05vh1FrL583JvggIUbcw)
> 
> Please also upload the post-processed facemesh image below. It seems necessary to check if the issue is occurring at a specific resolution.

512x768
![1701249318857](https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/63891627/7f606197-405d-454d-bd35-ca0120a9bbf9)

**Comment by ltdrdata** [comment]
> > > > Send me the your workflow and sample input.
> > > 
> > > 
> > > ![test-mediapie](https://private-user-images.githubusercontent.com/63891627/287141519-97cbc73c-c158-4780-aecb-3e51c06f5ed6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MDQ0NjgsIm5iZiI6MTcwMTQwNDE2OCwicGF0aCI6Ii82Mzg5MTYyNy8yODcxNDE1MTktOTdjYmM3M2MtYzE1OC00NzgwLWFlY2ItM2U1MWMwNmY1ZWQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDA0MTYwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4ZmMwZmNiMTk5ZWE5MzQ3NDhkMTcyYjYyZWQyODc5NTFiOGU2MTEyMDZiZDk5MWYzNDBiYjRhNmVlY2MxODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ylp4Uk5NoLvGy5D9eNakURF05vh1FrL583JvggIUbcw)
> > 
> > 
> > Please also upload the post-processed facemesh image below. It seems necessary to check if the issue is occurring at a specific resolution.
> 
> 512x768 ![1701249318857](https://private-user-images.githubusercontent.com/63891627/287146581-7f606197-405d-454d-bd35-ca0120a9bbf9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MzIyODAsIm5iZiI6MTcwMTQzMTk4MCwicGF0aCI6Ii82Mzg5MTYyNy8yODcxNDY1ODEtN2Y2MDYxOTctNDA1ZC00NTRkLWJkMzUtY2EwMTIwYTliYmY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDExNTk0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2YzY1MWNhZjY0M2IwYThkMTRkOThmMDUzMTQ3YTU0ZWJmYjdiZWJlMmRlMTE2MGQxMGQ2NjhjZjQyYTY5NzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sZBdvUw09ZqGfMJgyeOuJIrIzgOyqc8z-VAddxJVkzw)

This is fixed on V4.38
